source/geometries/Next100FieldCage.cc

```diff
@@ -186,12 +198,12 @@ void Next100FieldCage::SetMotherPhysicalVolume(G4VPhysicalVolume* mother_phys)
 void Next100FieldCage::Construct()
 {
   /// Calculate lengths of active and buffer regions
-  active_length_ = gate_cathode_centre_dist_ - grid_thickn_/2.;
+  active_length_ = GetELzCoord() + gate_teflon_dist_ + active_cathode_dist_ + cathode_thickn_/2. - grid_thickn_/2.;
```

Comment:
A length cannot depend on the relative position of the gate in its mother volume (`GetELzCoord()`). This will give you an in correct length of the active volume if it is not zero. I understand you debugged this geometry using the `Next100OpticalGeometry`, where that position is zero. Please, use the `Next100` geometry instead, and check all the dimensions and volume overlaps. You can comment the placement of the ICS for the moment and skip the overlaps of the vessel, since those volumes still need to be reviewed.

Reply:
Changed! Now there are no overlaps!